samples/android/src/main/java/cafe/adriel/voyager/sample/transition/TransitionActivity.kt

```diff
@@ -0,0 +1,165 @@
+package cafe.adriel.voyager.sample.transition
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.animation.AnimatedContentTransitionScope
+import androidx.compose.animation.ContentTransform
+import androidx.compose.animation.core.FiniteAnimationSpec
+import androidx.compose.animation.core.LinearEasing
+import androidx.compose.animation.core.keyframes
+import androidx.compose.animation.core.tween
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.animation.scaleIn
+import androidx.compose.animation.scaleOut
+import androidx.compose.animation.shrinkVertically
+import androidx.compose.animation.slideInHorizontally
+import androidx.compose.animation.slideInVertically
+import androidx.compose.animation.slideOutHorizontally
+import androidx.compose.animation.slideOutVertically
+import androidx.compose.animation.togetherWith
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.IntSize
+import cafe.adriel.voyager.core.screen.Screen
+import cafe.adriel.voyager.navigator.Navigator
+import cafe.adriel.voyager.navigator.isPop
+import cafe.adriel.voyager.navigator.isPush
+import cafe.adriel.voyager.transitions.ScreenTransition
+import cafe.adriel.voyager.transitions.ScreenTransitionContent
+
+class TransitionActivity : ComponentActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            Navigator(TransitionScreen) {
+                TransitionDemo(it)
+            }
+        }
+    }
+}
+
+@Composable
+fun TransitionDemo(
+    navigator: Navigator,
+    modifier: Modifier = Modifier,
+    content: ScreenTransitionContent = { it.Content() }
+) {
+    val transition: AnimatedContentTransitionScope<Screen>.() -> ContentTransform = {
+        // Define any StackEvent you want transition to be
+        val isPush = navigator.lastAction.isPush()
+        val isPop = navigator.lastAction.isPop()
+        // Define any Screen you want transition must be from
+        val isInvokerTransitionScreen = navigator.lastAction?.invoker == TransitionScreen
+        val isInvokerFadeScreen = navigator.lastAction?.invoker == FadeScreen
+        val isInvokerShrinkScreen = navigator.lastAction?.invoker == ShrinkScreen
+        val isInvokerScaleScreen = navigator.lastAction?.invoker == ScaleScreen
+        // Define any Screen you want transition must be to
+        val isTargetTransitionScreen = navigator.lastItem == TransitionScreen
+        val isTargetFadeScreen = navigator.lastItem == FadeScreen
+        val isTargetShrinkScreen = navigator.lastItem == ShrinkScreen
+        val isTargetScaleScreen = navigator.lastItem == ScaleScreen
```

Comment:
```suggestion
        val invoker = this.initialState
        val target = this.targetState
        // Define any StackEvent you want transition to be
        val isPush = navigator.lastEvent == StackEvent.Push
        val isPop = navigator.lastEvent == StackEvent.Pop
        // Define any Screen you want transition must be from
        val isInvokerTransitionScreen = invoker == TransitionScreen
        val isInvokerFadeScreen = invoker == FadeScreen
        val isInvokerShrinkScreen = invoker == ShrinkScreen
        val isInvokerScaleScreen = invoker == ScaleScreen
        // Define any Screen you want transition must be to
        val isTargetTransitionScreen = target == TransitionScreen
        val isTargetFadeScreen = target == FadeScreen
        val isTargetShrinkScreen = target == ShrinkScreen
        val isTargetScaleScreen = target == ScaleScreen
```

Reply:
I did not know about that API. I have checked, works as expected. Thank you!

Reply:
I would like if you can open a PR with this transitions examples using the `initialState` and `targetState` API to we have on our oficial Voyager samples.
Could be also the sample base that we could use to create a new Transition API for Screens and test on it.